File: LeetCode/Python Solutions/Time Needed to Buy Tickets/time.py
class Solution:
    def timeRequiredToBuy(self, tickets, k):
        time = 0
        for i, ticket in enumerate(tickets):
            if i <= k:
                time += min(ticket, tickets[k])
            else:
                time += min(ticket, tickets[k] - 1)
        return time

File: LeetCode/Java Solutions/Time Needed to Buy Tickets/time.java
class Solution {
    public int timeRequiredToBuy(int[] tickets, int k) {
        int time = 0;
        for (int i = 0; i < tickets.length; i++) {
            if (i <= k) {
                time += Math.min(tickets[i], tickets[k]);
            } else {
                time += Math.min(tickets[i], tickets[k] - 1);
            }
        }
        return time;
    }
}
